include/fuse_common.h

```diff
@@ -32,14 +32,22 @@ extern "C" {
 #endif
 
 /**
- * Information about open files
+ * Information about an open file handle
+ *
+ * File Handles are created by the open, opendir, and create methods and closed
+ * by the release and releasedir methods.  Multiple file handles may be
+ * concurrently open for the same file.  Generally, a client will create one
+ * file handle per file descriptor, though in some cases multiple file
+ * descriptors can share a single file handle.
  */
 struct fuse_file_info {
 	/** Open flags.	 Available in open() and release() */
 	int flags;
 
 	/** In case of a write operation indicates if this was caused by a
-	    writepage */
+	    delayed write from the page cache.  If so, then the context's pid,
+	    uid, and gid fields will not be valid, and the file handle may not
+	    correspond to the file descriptor responsible for the write. */
 	unsigned int writepage : 1;
```

Comment:
That's much better, thanks! I'd still like to omit the "and the file handle may not correspond to the file descriptor responsible for the write", because FUSE doesn't know anything about file descriptors.

Reply:
fuse_kernel.h says "file handle is guessed", but I was trying to come up with something more precise.  I think it's important to be precise here, especially since the relationship between file handles and file descriptors is so close.  It's like the difference between process IDs and thread IDs; on Linux they're almost, but not quite, identical.  So it's very important to carefully document the distinctions or people _will_ get them confused.

Reply:
I agree with the goal, but I don't think that one or two extra sentences in the comments are helping with this. I coherent, multi paragraph explanation somewhere else would be a different story...

Reply:
Where then?

Reply:
Maybe create a new doc/writeback.rst file for that? That would be a good place to also store other information related to that.